### PR TITLE
Authentication and profile

### PR DESCRIPTION
**Pull request by Jay-Gadhiya**
link: https://video-library-8vx9cjh2j-jay-gadhiya.vercel.app/

Feature added: login, signup, use guest login, logut
Profile : logut option

### CODE REVIEW

**Review by jeetbhalani15 — APPROVED** [review]
Auth context and all API calls works fine
- Code is readable & clean
- Auth UI looks good

File: src/Pages/Authentication/Login.jsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    // login click handler
+    const loginClickHandler = async (e) => {
+        e.preventDefault();
+        try {
+            const response = await axios.post("/api/auth/login", userData);
+            
+            localStorage.setItem("token", response.data.encodedToken);
+            authDispatch({ type : "USER_LOGIN", payload : response.data.encodedToken })
```
Comment (Kevin-Solomon) [inline src/Pages/Authentication/Login.jsx:40]: We can also store userDetails from this response.
```response.data.foundUser``` for Login
```response.data.createdUser``` for Sign Up

File: src/Pages/Authentication/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            navigate("/");
+
+        } catch (error) {
```
Comment (Kevin-Solomon) [inline src/Pages/Authentication/Login.jsx:43]: Catch Block only returns something if the ```response.status``` is 500.
We can check for other response status like user already exist using conditionals.

File: src/Components/Profile/profile-info.jsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { useAuth } from "../../context/authentication-context"
+
+export const ProfileInfo = () => {
+
+    return (
+        <div className="content-show">
+            <p><strong>Username</strong> : <span>Adarsh Balika</span></p>
+            <p><strong>Email</strong> : <span>adarshbalika@gmail.com</span></p>
+        </div>
```
Comment (Kevin-Solomon) [inline src/Components/Profile/profile-info.jsx]: Here you can display user info from context instead of hard coding it.

**Review by Kevin-Solomon — APPROVED** [review]

**Pull request merged**